### PR TITLE
boards: stm32: Convert boards I2C pins to devicetree

### PR DESCRIPTION
**Pull request by erwango**
Get STM32 boards configuring I2C using device tree and remove configuration from pinmux.c.

I had to tweak pinctrl on few boards in which configuration was incorrect/not coherent:
nucleo_f030r8: i2c2 was configured in PA11/PA12 non existing in package. Moved to PB10/PB11
nucleo_l432kc: i2c1 node was missing in dts. Added
nucleo_l452re: i2c1 node was missing in dts. Added
stm32f429i_disc1: i2c1 and i2c2 node were missing in dts. Added

### CODE REVIEW

File: boards/arm/google_kukui/pinmux.c [hunk 1]
```diff
@@ -13,14 +13,6 @@
 #include <pinmux/stm32/pinmux_stm32.h>
 
 static const struct pin_config pinconf[] = {
```
Comment (galak) [inline boards/arm/google_kukui/pinmux.c:15]: Do we need to keep this file now that `pinconf[]` is empty?

**Review by galak — APPROVED** [review]
LGTM, matches what my script produced.  Only question is related to the 2 boards that pinconf array is now empty.

**Review by gmarull — APPROVED** [review]
LGTM, I'd also start deleting empty pinmux.c files.

File: boards/arm/google_kukui/pinmux.c (hunk 1 above)
Comment (erwango) [reply boards/arm/google_kukui/pinmux.c:15]: I chose to keep it in order not to block board peripheral additions during the transition.
Of course we should avoid addition of peripherals that should be handled in dts.

**Review by lochej — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Comment by galak** [comment]
@erwango added a commit to bump the number of bk builders so we can see if this passes CI.

**Comment by galak** [comment]
@erwango merged the hal change if you want to fixup the west.yml commit now.

**Comment by erwango** [comment]
@galak updated the west.yaml and removed the BK patch

**Comment by galak** [comment]
@erwango can I get you to rebase since I merged the ADC PR.

**Comment by erwango** [comment]
> @erwango can I get you to rebase since I merged the ADC PR.

@galak sure, here you go.

File: boards/arm/google_kukui/pinmux.c (hunk 1 above)
Comment (jackrosenthal) [reply boards/arm/google_kukui/pinmux.c:15]: OK to remove this file from CODEOWNERS perspective on this one :)

Will let you make the call @erwango

**Review by jackrosenthal — APPROVED** [review]

**Comment by galak** [comment]
Merging as this passed CI before.

**Pull request merged**